src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,11 +1,42 @@
-export const TodoFilter = () => (
+import { ChangeEvent } from 'react';
+
+import { FilterOption } from '../../enums/filter-options';
+
+type Props = {
+  filterOption: FilterOption;
+  searchQuery: string;
+  onFilterTodos: (event: ChangeEvent<HTMLSelectElement>) => void;
+  onSearch: (event: ChangeEvent<HTMLInputElement>) => void;
+  onResetSearch: () => void;
+};
+
+const filterOptionsArray = Object.values(FilterOption);
+
+export const TodoFilter: React.FC<Props> = ({
+  filterOption,
+  searchQuery,
+  onFilterTodos,
+  onSearch,
+  onResetSearch,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
+        <select
+          data-cy="statusSelect"
+          value={filterOption}
+          onChange={onFilterTodos}
+        >
+          {filterOptionsArray.map(option => {
+            const optionTitle =
+              option.charAt(0).toUpperCase() + option.slice(1);
+
+            return (
+              <option value={option} key={option}>
+                {optionTitle}
+              </option>
+            );
```

Comment:
It's not a good practice to generate keys on render. It can lead to performance issues and unexpected behavior in React. In this case, the `option` value is not likely to change, so it's okay to use it as a key. But keep this in mind for future reference. ([here](https://medium.com/blackrock-engineering/5-common-mistakes-with-keys-in-react-b86e82020052) is why)